src/pkg/path/drive.go

```diff
@@ -38,3 +38,13 @@ func GetDriveFolderPath(p Path) (string, error) {
 
 	return Builder{}.Append(drivePath.Folders...).String(), nil
 }
+
+// FormatDriveFolders takes a driveID and a set of unescaped element names,
+// including the root folder, and returns a *path.Builder containing the
+// canonical path representation for the drive path.
+func FormatDriveFolders(
```

Comment:
Or, if you want to be more verbose, `BuilderWithDriveFolders`.  `Format` feels like a direct transformation on the provided values more so than a constructor.
```suggestion
func BuildDriveLocation(
```